### ISSUE TITLE
`onProgress` `percentage` type wrong

### ISSUE THREAD

**Issue by vincerubinetti** [ibrod83/nodejs-file-downloader#68]
The type definition says `percentage` is a `string`, but it can also be `NaN`, so the type should be `string | number`.

On a related note, I find it weird that percent is a `toFixed`-stringified version of the number rather than just a number. This would make the type more consistently and expectedly just `number`. Also I find it a little unexpected that the range of it is 0 to 100 instead of 0 to 1, but that's more of a preference.

If you ever making a breaking change to this library, these changes would be welcome.

**Comment by ibrod83** [comment]
You're right, that was some stupid misunderstanding on my part. As you said, this is breaking change, so perhaps in the future
